PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -826,7 +830,7 @@ class CustomFileDialog::Impl
 
 ///////////////////////////////////////////////////////////////////////////////
 
-CustomFileDialog::CustomFileDialog(HWND hwnd) : _impl{std::make_unique<Impl>()}
+CustomFileDialog::CustomFileDialog(HWND hwnd) : _impl{ std::make_unique<Impl>() }
```

Comment:
Strangely, MSVC (GCC and Clang do not) generates different code depending on if you brace or parantheses-initializer. Also, generates different code if you use `new` instead of `std::make_unique`.

https://godbolt.org/z/3WfKaqEYP

Reply:
I don't see any significant differences between braces vs parentheses init. Below is a diff of your example when I left only int overload and only float overload:
![image](https://user-images.githubusercontent.com/9664141/120885164-29db6880-c5f0-11eb-8e55-b0f329d7d6ef.png)

As for new, yes, it would be different as expected. 
But still, using `make_unique` over bare `new` is a good practice to avoid leaks if constructor throws:
https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique

Reply:
I don't disagree, I was just surprised at the difference as I'd expected the optimizer to reduce them to the same thing in the end.

Don't look at the `wrapper_type`s as they are `__forceinline`d - look at the outputs for `func0`/`1`/`2`.